yt_dlp/extractor/mx3.py

```diff
@@ -0,0 +1,98 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import urlhandle_detect_ext
+from ..networking import HEADRequest
+
+
+class Mx3IE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?mx3\.ch/t/(?P<id>[0-9A-Za-z]+)'
+    _TESTS = [{
+        'url': 'https://mx3.ch/t/1Cru',
+        'md5': '82510bf4c21f17da41bff7e1ffd84e78',
+        'info_dict': {
+            'id': '1Cru',
+            # This one is audio-only. It's a mp3, but we have to make a HEAD request to find out.
+            'ext': 'mp3',
+            'artist': 'Tortue Tortue',
+            'genre': 'Rock',
+            'thumbnail': 'https://mx3.ch/pictures/mx3/file/0101/4643/square_xlarge/1-s-envoler-1.jpg?1630272813',
+            'title': 'Tortue Tortue - S\'envoler',
+        }
+    }, {
+        'url': 'https://mx3.ch/t/1LIY',
+        'md5': '4117489dff8c763ecfbb0b95a67d6c8e',
+        'info_dict': {
+            'id': '1LIY',
+            # This is a music video. 'file' says: ISO Media, MP4 Base Media v1 [ISO 14496-12:2003]
+            'ext': 'mp4',
+            'artist': 'The Broots',
+            'genre': 'Electro',
+            'thumbnail': 'https://mx3.ch/pictures/mx3/file/0110/0003/video_xlarge/frame_0000.png?1686963670',
+            'title': 'The Broots-Larytta remix "Begging For Help"',
+        }
+    }, {
+        'url': 'https://mx3.ch/t/1C6E',
+        'md5': '1afcd578493ddb8e5008e94bb6d97e25',
+        'info_dict': {
+            'id': '1C6E',
+            # This one has a download button, yielding a WAV.
+            'ext': 'wav',
+            'artist': 'Alien Bubblegum',
+            'genre': 'Punk',
+            'thumbnail': 'https://mx3.ch/pictures/mx3/file/0101/1551/square_xlarge/pandora-s-box-cover-with-title.png?1627054733',
+            'title': 'Alien Bubblegum - Wide Awake',
+        }
+    }]
+
+    def _real_extract(self, url):
+        track_id = self._match_id(url)
+        webpage = self._download_webpage(url, track_id)
```

Comment:
Actually, looking around the page, there is a bunch more optional data on the webpage that isn't in the JSON.
If you like, you can also try to extract those.
`Year of creation` -> `release_year`
`single-stats` -> `view_count`
`Added on` -> `upload_date`
`Description` -> `description` (not always present)
`Tag` -> `tags` (not always present)
Maybe there is more I didn't see.

Reply:
Who consumes all this information, though? Descriptions and tags sound useful enough to extract, maybe I'll add those, but they won't be embedded into the downloaded MP3s or anything by default, right?

Reply:
Some of this info would be embedded if you use `--embed-metadata`.
Some info may also be useful if you want to filter certain downloads, say you wanted to only download tracks uploaded within the last month.
If you don't think it is worth the hassle to implement, you don't need to do this.

Reply:
I see; I've added the date now from the HEAD request because it was easy. Yes I think it's diminishing returns here, it looks like we'd have to hunt those down individually in the HTML using nearby translatable strings as an anchor. Seems pretty fragile, too.